schemas/styles/csl-variables.rnc

```diff
@@ -74,6 +74,7 @@ div {
     | "authority"
     | "call-number"
     | "citation-label"
+    | "citekey"
```

Comment:
```suggestion
```

I don't understand why we need this in csl styling, when citation-label was intended for this case. Here's the definition from the spec:

> label identifying the item in in-text citations of label styles (e.g. "Ferr78"). May be assigned by the CSL processor based on item metadata

As in, adding `citekey` to the input data simply means that if the CSL processor does not assign, then it uses `citekey`.

Right?

Reply:
Or is idea a style would need to test for it?

Reply:
No, these really are completely different things. Ferr78 is not the citekey looked up in a data file, but rather a formatted citation in a style used primarily in math, statistics, and engineering.

The reason that a _style_ might have the citekey variable would be to facilitate transfer between GUI and plain text citation systems. For example, to go from a Word document with Zotero or Mendeley citations to an a document edited using pandoc, a style could be written that converts formatted citations into citekeys, affixes, and locators in the pandoc style.

This could be accomplished by instructing applications and processors to include a function to "unformat" citations, but a style is simple application-agnostic approach to that as well.

Reply:
> Ferr78 is not the citekey **looked up in a data file**, but rather a formatted citation in a style used primarily in math, statistics, and engineering.

That's not what I mean above, which is only from the standpoint of the _formatted output_. 

From that standpoint, whether generated or not, the label is (has only ever been intended to be) a local human-readable token id for the reference item; eg:

```
foo bar [doe99]
```

... that then matches the equivalent in the bibliography:

```
[doe99] Doe, J ...
```

It's just, in a style with that output, the label is not (necessarily) generated.

So I'm still not understanding ;-)

If we add this to `cs:style`, we need a clear and simple explanation how it really differs from `citation-label`.

Reply:
> we add this to `cs:style`, we need a clear and simple explanation how it really differs from `citation-label`, and maybe to change the definition of the latter.

Also, if we need both, maybe should use the same naming convention; say `citation-key`?

Not sure; just a thought.

Reply:
PS - some history:

https://discourse.citationstyles.org/t/citation-label/761

Reply:
Yes, I think we are agreeing that `citekey` and `citation-label` are not intended to mean the same thing at all.

The question we are having is whether the data-input-related `citekey` should be available for rendering in styles or only in the data and citation json schemas. I'm saying there is a specific use case for having it in styles: to facilitate transfer between GUI and text-based editors. 

I think we can make this clear in the documentation.